### PR TITLE
Suggested additions/changes to Categorize/EHR

### PR DESCRIPTION
**Pull request by enricoferrero**
OK, here's an attempt at incorporating #342 in the EHR section of Categorize. I think it can stand as its own paragraph, or it can be somehow amalgamated with the previous one (note that this was only very slightly edited).

I would also like to propose a few changes to the description of #63. Please feel free to disagree and revert back to the original :smiley:

Feedback very welcome!

### CODE REVIEW

File: sections/03_categorize.md [hunk 1]
```diff
@@ -168,19 +181,20 @@ showed that unsupervised feature construction of many features via denoising
...
-feature construction methods [@doi:10.1038/srep26094].  Razavian et al.
-[@arxiv:1608.00647] used a set of 18 common lab tests to predict disease onset
-using both CNN and LSTM architectures and demonstrated and improvement over baseline
-regression models. However, numerous challenges including data integration (patient
-demographics, family history, laboratory tests, text-based patient records,
-image analysis, genomic data) and better handling of streaming temporal data
-with many features, will need to be overcome before we can fully assess the
-potential of deep learning for this application area.
+features being useful for subtyping within a single disease. In a concurrent
+large-scale analysis of EHR data from 7000,000 patients, Miotto et al
```
Comment (cgreene) [inline sections/03_categorize.md]: I think it's 700,000

**Comment by cgreene** [comment]
A bit swamped today. Made just a quick scan & think the number is 700k if I recall correctly.

File: sections/03_categorize.md (hunk 1 above)
Comment (enricoferrero) [inline sections/03_categorize.md]: Yes :smile:

File: sections/03_categorize.md [hunk 2]
```diff
@@ -142,21 +142,34 @@ This indicates a potential strength of deep methods. It may be possible to
...
+[@doi:10.1145/2939672.2939823], who built embeddings using a multi-layer network
+architecture which mimics the structure of claims data. While promising,
+difficulties in evaluating the quality of these kinds of features and variations
+in clinical coding practices remain as challenges to using them in practice.
+
+NLP has also been applied directly to EHRs to predict disease comorbidities and
+phenotypes as well as novel gene - disease associations. Gligorijevic et al
+[@doi:10.1038/srep32404] built neural embedding models and mined more than 35
+million records collected over a decade. The learned disease representations
+significantly ouperformed state-of-the-art methods, achieving close to 86%
```
Comment (agitter) [inline sections/03_categorize.md]: This is outside my area and I haven't read the Gligorijevic paper, but I am curious whether 86% accuracy should be considered good or good enough to be useful in this domain.  In some domains that might be useful.  In others it would be quite poor even if better that existing methods.

Comment (cgreene) [inline sections/03_categorize.md]: @brettbj can you comment?

Comment (enricoferrero) [inline sections/03_categorize.md]: @agitter My impression is that 86% accuracy is pretty good for this type of task given the complexity/heterogeneity of EHR data. It might be worth looking at Table 1 in the paper: they provide precision@K values for theirs and other methods and the performance improvement is often quite impressive. I think in information retrieval, precision@K might be a better performance metric than accuracy (but accuracy easier to report being a single number).

Comment (agitter) [inline sections/03_categorize.md]: @enricoferrero thanks for clarifying.  I'm not sure that we need to expand on this in the paragraph, but I'm always curious how close to "solved" a problem is even after substantially improving on previous state of the art.

Comment (agapow) [inline sections/03_categorize.md]: The issue of numbers occurs elsewhere in the paper (at least in the parts I wrote). I think that the number itself is not so much significant as is whether it improves on the prior state of the art and by how much. (And perhaps whether it "solves" the problem.) In a review paper for a general audience, I'd suggest that numbers might be superfluous except as illustrations.

**Comment by agitter** [comment]
Merging #360 introduced conflicts here.

File: sections/03_categorize.md (hunk 2 above)
Comment (brettbj) [inline sections/03_categorize.md]: Not sure if I'm reading something incorrectly but:
For this they only look at ICD9 codes though (so wouldn't have as many issues with noise/ heterogeneity etc.) 

The task is:
Disease Phenotyping: Identifying diseases with similar contexts as the query disease.

But it looks like they define these as having shared GWAS hits (at 10^-5 significance)

Here's how it looks like they determine that:
Specifically, we selected 2,739 diseases found in the GWAS data and for each retrieved K nearest diseases, with k in {1, 2, 5, 10} . Each of the retrieved diseases was labeled as positive if it shares a gene with the query disease, and labeled negative otherwise, which is used as a proxy for having the same phenotype54. Then, we computed precision@K for each disease as a fraction of positive neighbors within the K retrieved ones, and report the average precision over all 2,739 diseases.

The selection of diseases is really important, for example I'd be really suspicious of any cancers etc. as they are likely to share genes? They also include things like 234 forms of Tuberculous and tuberculosis - both are caused by the same bacteria... is it novel to predict these are related? (that's almost 10%)

In any event it makes sense that predicting diseases that co-occur have similar genetic factors. I'm not sure why this method does so much better than the others tested but it is not surprising that a method is able to do well.

Comment (enricoferrero) [inline sections/03_categorize.md]: @brettbj Thanks for reviewing the paper and feedback.

My understanding is that the authors try to address two tasks:

1.  Disease2Disease (D2D, or disease phenotyping), where the model is trained exclusively using raw EHR data and learn the disease representations using a method similar to word2vec where each record is a sentence and each disease is a word:
    > In this section we propose the disease2diseases (D2D) approach for learning disease representations, building upon ideas introduced by the recently proposed word2vec algorithm[...] the language model learns representations of diseases in a low-dimensional space using each patient dis- charge record as a “sentence” and the diseases within the record as “words”.

    To _validate_ their model they then use shared GWAS genes to asses how close two diseases really are (but the GWAS data is _not_ used during training): 
    > To train the D2D model, we used only EHR data (P dataset) [...] Specifically, we selected 2,739 diseases found in the GWAS data and for each retrieved K nearest diseases, with ∈K {1,2,5, 10}. Each of the retrieved diseases was labeled as positive if it shares a gene with the query disease, and labeled negative otherwise, which is used as a proxy for having the same phenotype54. Then, we computed precision@K for each disease as a fraction of positive neighbors within the K retrieved ones, and report the average precision over all 2,739 diseases...

2. DiseaseAndGenes2Diseases (DAG2D, or retrieval of disease - gene associations), a model using GWAS data to learn representations of diseases and genes:
    > The DAG2D model assumes that a subset of diseases from the training data D are associated with genes, where
the associations are provided by domain experts and considered as domain knowledge. We leverage this informa- tion by assigning a vector representation to each gene, and make use of disease contexts in the training data to jointly learn both disease vectors and gene vectors in the same low-dimensional space.

    This model is simply validated on a held-out test set:
    > To train the DAG2D model (for both of the tasks), we extended patient discharge records by associ- ating diseases to corresponding genes according to GWAS data [...] We randomly selected 20% of the diseases that have associated genetic data (the diseases found in the GWAS data set) and removed all their gene associations from the training data. Although we removed genetic information, these diseases are not removed from the original EHR data, so that we are able to learn their vector representations. Second, we learn DAG2D on the data set where the remaining 80% of diseases remained associated with their corresponding gene information data. The DAG2D model then contains low-dimensional representations of diseases and genes in the same embedding space, and we evaluated model performance by measuring how many of the removed genetic associations were correctly retrieved

So, given the performance measures and comparisons they provide I think the results of both models are quite impressive: in D2D they recapitulate genetics-based distances between diseases only using EHR data and in DAG2D they convincingly recover known disease genetic associations from data held out during training.

I'd be happy to try and improve my summary if it's unclear: any specific suggestion would be welcome/helpful.

@agapow Sure, I can remove the accuracy figure if that's uninformative. Do we have a paper-wide policy on this?

CC @cgreene @agitter

Comment (agapow) [inline sections/03_categorize.md]: It's an overall style thing and my read of the journal guidelines. I'll leave it to @cgreene @agitter to make the call.

Comment (agitter) [inline sections/03_categorize.md]: We don't currently have a paper-wide policy.  There are other sub-sections that include performance and accuracy values.

I'd still like to wait for a full review from @cgreene before we merge.  I suggest that we don't delay merging because of the accuracy statistic.  We will need to set a paper-wide and can apply it universally during editing.

@agapow can you please refer me to the relevant journal guidelines that you found?

Comment (agapow) [inline sections/03_categorize.md]: http://rsif.royalsocietypublishing.org/content/information-authors-0

The key quote being:

> These articles should put such research in a wider context and be written in a style that will make them accessible to readers in a wide range of disciplines.

Comment (brettbj) [inline sections/03_categorize.md]: @enricoferrero  - The issue I have is that I don't think finding these "diseases" have a common SNP (not even genome-wide significance) is interesting or impressive. What would a classifier that just looked for a common word get....? Or a classifier that just picked the closest ICD-9 code by number? The definition of different disease is flawed in my eyes. They looked for two ICD-9 codes with a common SNP, oftentimes these ICD-9 codes are a single disease presenting in a different part of the body. This is why phenotyping algorithms are usually used.

Check out the supplement: https://www.nature.com/article-assets/npg/srep/2016/160831/srep32404/extref/srep32404-s1.csv

 
image

or 

image

or

image

or 

image

or 

image

Comment (cgreene) [inline sections/03_categorize.md]: I agree with @brettbj that the results - once considering the predictions from the supplement - are a bit uninspiring. The results support that this algorithm is better than some other algorithms for this specific task, but these do seem to be trivial predictions. An evaluation that focused on only items that couldn't be found by e.g. a shared word in the disease name might show that the method is still compelling. I think it's fine to mention the work, but I am a bit wary of trumpeting the results as the arrival of deep networks in this space. Let me try to make some revisions (we need to make a merge commit anyway) to add more nuance to our description, and let's see what people think.

**Comment by cgreene** [comment]
@enricoferrero / @brettbj : see what you think of the aggregate changes now with my revisions. I tried to couch them a bit more in the difficult to eval sense (we may need to do this at other places too as we revise). The basic message is refined to: promising but needs more comparison to methods that could be used for these problems.

**Comment by brettbj** [comment]
LGTM

File: sections/03_categorize.md (hunk 2 above)
Comment (enricoferrero) [inline sections/03_categorize.md]: @brettbj @cgreene thanks, that's fair enough - I admittedly didn't look in the supplementary material.

File: sections/03_categorize.md [hunk 3]
```diff
@@ -143,21 +143,26 @@ This indicates a potential strength of deep methods. It may be possible to
...
+[@doi:10.1145/2939672.2939823]. Methods for neural embeddings learned from
+electronic health records have at least some ability to predict disease-disease
+associations [@doi:10.1038/srep32404]. However, the evaluation did not
+differentiate between simple predictions (i.e. the same disease in different
+sites of the body) and non-intutive ones. While promising, a lack of rigorous
+evaluations of the real-world utility of these kinds of features makes
+current contributions in this area difficult to evaluate. To examine the true
+utility, comparisons need to be performed against leading approaches (i.e.
+algorithms and data) as opposed to simply evaluating multiple algorithms on
+the same potentially limited dataset.
```
Comment (enricoferrero) [inline sections/03_categorize.md]: @cgreene, I think your summary looks good - my only concern is that the second model (DAG2D) in #342 is not even mentioned anymore. To me, that is the most interesting part of the paper: being able to identify disease-associated genes from EHR data has big implications for drug discovery (and specifically for target identification). It might be that DAG2D suffers from the same flaws of D2D but my preference would have been to mention this particular application (which doesn't seem to be explored elsewhere).
CC @brettbj @agitter

File: sections/03_categorize.md [hunk 4]
```diff
@@ -143,21 +143,26 @@ This indicates a potential strength of deep methods. It may be possible to
...
-have the natural document structure of clinical notes, and this issue was
-addressed by E. Choi et al. [@doi:10.1145/2939672.2939823], who built
-embeddings using a multi-layer network architecture which mimics the structure
-of claims data. While promising, difficulties in evaluating the quality of
-these kinds of features and variations in clinical coding practices remain as
-challenges to using them in practice.
+Several authors have created reusable feature sets for medical terminologies
+using natural language processing (NLP) and neural embedding models, as
+popularized by Word2vec [@tag:Word2Vec]. A goal of learning terminologies
+for different entities in the same vector spaec is to find relationships
```
Comment (enricoferrero) [inline sections/03_categorize.md]: Typo, spaec --> space

**Comment by cgreene** [comment]
@enricoferrero - feel free to file a new PR if you want to do more wordsmithing on it. Thanks!
